diff --git a/dmenu.c b/dmenu.c
index 65f25ce..d621f06 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -354,7 +354,9 @@ keypress(XKeyEvent *ev)
 			while (cursor > 0 && !strchr(worddelimiters, text[nextrune(-1)]))
 				insert(NULL, nextrune(-1) - cursor);
 			break;
+		case XK_x: /* paste selection */
 		case XK_y: /* paste selection */
+		case XK_X: /* paste selection */
 		case XK_Y:
 			XConvertSelection(dpy, (ev->state & ShiftMask) ? clip : XA_PRIMARY,
 			                  utf8, utf8, win, CurrentTime);
